src/xml/ConfigParser.cpp

```diff
@@ -117,10 +117,10 @@ void OnFatalErrorFunc(void *userData, const char *error, ...)
 
 precice::logging::Logger ConfigParser::_log("xml::XMLParser");
 
-ConfigParser::ConfigParser(const std::string &filePath, const ConfigurationContext &context, std::shared_ptr<precice::xml::XMLTag> pXmlTag)
+ConfigParser::ConfigParser(std::string_view filePath, const ConfigurationContext &context, std::shared_ptr<precice::xml::XMLTag> pXmlTag)
     : m_pXmlTag(std::move(pXmlTag))
 {
-  readXmlFile(filePath);
+  readXmlFile(std::string(filePath));
```

Comment:
Why does readXmlFile need an owning string in the interface?

Same below

Reply:
Yes, your point is reasonable. 

The reason I was doing an explicit call at the interface was to make it compatible with std::ifstream here. 

https://github.com/precice/precice/blob/a8d2b233c640bdd86f67f036313bed5aeda4038e/src/xml/ConfigParser.cpp#L172

But it is not a clean way of passing it to the interface like this.